### PR TITLE
add num_epochs to train_step calculation

### LINKED ISSUE #137: Warmup steps calculation should not ignore epochs.
The warmup steps calculation should not ignore epochs. See here:

https://github.com/huggingface/setfit/blob/7735e8e3b208edb8dfb549beb16e585453c5f44e/src/setfit/trainer.py#L275

The bug is where train steps is calculates here:

https://github.com/huggingface/setfit/blob/7735e8e3b208edb8dfb549beb16e585453c5f44e/src/setfit/trainer.py#L267

the right implementation is here (but not always executed):

https://github.com/huggingface/setfit/blob/7735e8e3b208edb8dfb549beb16e585453c5f44e/src/setfit/trainer.py#L253

I can provide a PR.

### PR DESCRIPTION
**Pull request by PhilipMay**
fixes #137

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
Nice catch, thanks for the fix!
